cmd/influxd/launcher/query_test.go

```diff
@@ -745,3 +748,79 @@ from(bucket: "%s")
 		t.Fatal(err)
 	}
 }
+
+func TestLauncher_Query_PushDownWindowAggregateCount(t *testing.T) {
+	l := launcher.RunTestLauncherOrFail(t, ctx, "--feature-flags", "pushDownWindowAggregateCount=true")
+	l.SetupOrFail(t)
+	defer l.ShutdownOrFail(t, ctx)
+
+	l.WritePointsOrFail(t, `
+m0,k=k0 f=0i 0
+m0,k=k0 f=1i 1000000000
+m0,k=k0 f=2i 2000000000
+m0,k=k0 f=3i 3000000000
+m0,k=k0 f=4i 4000000000
+m0,k=k0 f=5i 5000000000
+m0,k=k0 f=6i 6000000000
+m0,k=k0 f=5i 7000000000
+m0,k=k0 f=0i 8000000000
+m0,k=k0 f=6i 9000000000
+m0,k=k0 f=6i 10000000000
+m0,k=k0 f=7i 11000000000
+m0,k=k0 f=5i 12000000000
+m0,k=k0 f=8i 13000000000
+m0,k=k0 f=9i 14000000000
+m0,k=k0 f=5i 15000000000
+`)
+
+	query := fmt.Sprintf(`
+from(bucket: "%s")
+	|> range(start: 1970-01-01T00:00:00Z, stop: 1970-01-01T00:00:15Z)
+	|> aggregateWindow(every: 5s, fn: count)
+	|> drop(columns: ["_start", "_stop"])
+`, l.Bucket.Name)
+	res := l.MustExecuteQuery(query)
+	defer res.Done()
+	got := flux.NewSliceResultIterator(res.Results)
+	defer got.Release()
+
+	dec := csv.NewMultiResultDecoder(csv.ResultDecoderConfig{})
+	data := `
+#datatype,string,long,long,string,string,string,dateTime:RFC3339
+#group,false,false,false,true,true,true,false
+#default,_result,,,,,,
+,result,table,_value,_field,_measurement,k,_time
+,,0,5,f,m0,k0,1970-01-01T00:00:05Z
+,,0,5,f,m0,k0,1970-01-01T00:00:10Z
+,,0,5,f,m0,k0,1970-01-01T00:00:15Z
+`
```

Comment:
Do we maybe wanna consider writing this test so that it can run multiple test cases? You could even just merge as is and I can do it as part of registering the bare aggregate.

Reply:
I refactored this, but I refactored it mostly to test around all aggregate window calls. I think we can further refactor this to share it amongst the code for the other tests though.

I didn't notice you mentioned bare aggregates until I was already done. Do you want me to refactor it further?

Reply:
No, I can work on the further refactor as part of my PR.